### ISSUE TITLE
AutoMapper valueObjectToObj return {} for simple object

### ISSUE THREAD

**Issue by GaetanCottrez** [4lessandrodev/rich-domain#183]
**Describe the bug**
If an entity contains an array of simple objects, the automapper returns an empty object ({}) instead of the object.

**To Reproduce**

I noticed this issue in the case I am about to describe, but I think that an array of simple objects in an entity must present the same problem.

I have two entities with the following properties:
```ts
export interface EntityAProps {
	foo: string;
	entitiesB: EntityB[]
}

export interface EntityBProps {
	bar: string;
	arraySimpleObject: SimpleObject[]
}

export interface SimpleObject {
	value1: string;
	value2: string;
	value3: number;
	value4: Date;
	value5: string[];
}
```

To summarize, `EntityA` contains an array of `EntityB`, which in turn contains an array of simple objects.

If we do a `toObject` on `EntityA`, `EntityB` is correctly transformed into an object, but `arraySimpleObject` will contain empty objects `{}`.

The cause is due to `valueObjectToObj` in `AutoMapper`. The case described does not fit into any of the handled cases.

`let result = {} as { [key in keyof Props]: any };` is never modified, and the function returns `return result as any;`.

I suggest adding `this.validator.isObject(entity)` to the possible cases for the `isSimpleValue` variable (as in the `valueObjectToObj` method).

After adding this validator, the issue no longer occurs.

I will create a pull request to fix it.

And, of course, we'll need to bump the version of `rich-domain` for `types-ddd`!

**Lib Version**
rich-domain v1.23.3

**Comment by 4lessandrodev** [comment]
@GaetanCottrez  Thank you for reporting this issue and for the detailed explanation! I also appreciate your proposed solution and your willingness to create a pull request to address the problem. Contributions like this are incredibly valuable to the continuous improvement of the project. I’ll review your PR as soon as it’s available, and we’ll make sure to bump the rich-domain version as needed.

Thanks again for your support and for helping to make the project more robust!

**Comment by GaetanCottrez** [comment]
Thank you @4lessandrodev !

quick and efficient in handling Pull Request !

Do you want me to create an issue and/or a pull request on the types-ddd repository to bump the version of rich-domain as well?

**Comment by 4lessandrodev** [comment]
Yes, please feel free to create an issue and/or pull request on the types-ddd repository to bump the rich-domain version. I truly appreciate your help with this.

Also, I’m currently publishing the new version with the fix, 1.23.4. Thanks again for your support!

**Comment by GaetanCottrez** [comment]
> Yes, please feel free to create an issue and/or pull request on the types-ddd repository to bump the rich-domain version. I truly appreciate your help with this.
> 
> Also, I’m currently publishing the new version with the fix, 1.23.4. Thanks again for your support!

Ok, i created a [pull request](https://github.com/4lessandrodev/type-ddd/pull/444) for bump the last version of rich-domain in types-ddd.

I'll let you organize a release of types-ddd because I don't have control over it.
